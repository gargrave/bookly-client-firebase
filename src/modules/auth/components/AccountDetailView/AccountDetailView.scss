@import '../../../../styles/_views';

.accountDetailView {
  @include viewWrapper;
}
