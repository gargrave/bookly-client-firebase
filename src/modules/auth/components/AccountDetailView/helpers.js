// @flow
import { format } from 'date-fns';

import type { User } from '../../flowtypes';
import type { Profile } from '../../../profiles/flowtypes';

export const userBasicDetailsTextList = (user: User, profile: Profile) => {
  const {
    email,
    emailVerified,
  } = user;
  const {
    firstName,
    lastName,
  } = profile;

  const nameStr = `${firstName} ${lastName}`;
  const verifiedStr = emailVerified ? '' : ' (unverified)';
  const emailStr = `${email}${verifiedStr}`;

  return [
    { title: 'Name', value: nameStr },
    { title: 'Email', value: emailStr },
  ];
};

export const userRegDetailsTextList = (user: User) => {
  const {
    lastLogin,
    registered,
  } = user;

  return [
    { title: 'Registered', value: format(registered, 'MMM. DD, YYYY, HH:mm:ss') },
    { title: 'Last login', value: format(lastLogin, 'MMM. DD, YYYY, HH:mm:ss') },
  ];
};
