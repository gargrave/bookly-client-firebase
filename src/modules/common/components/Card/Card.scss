@import '../../../../styles/_shadows';

$textColor: #6e6969;
$titleColor: #5d5656;

.card {
  @include boxShadowMed;
  background-color: #f9f9f9;
  border: solid 1px #bbb;
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
}

.hoverableCard:hover {
  background: #e4effc;
  cursor: pointer;
}

.header,
.title,
.text {
  margin: 0;
}

.header,
.title {
  color: $titleColor;
  font-weight: bold;
}

.header {
  font-size: 1.4rem;
}

.detailCard {
  padding: 15px;

  .title {
    font-size: 1.3rem;
    line-height: 2.5rem;
  }

  .title + .cardText {
    font-size: 1.1rem;
  }

  .cardText + .cardText {
    margin-top: 8px;
  }
}

.listDetailCard {
  padding: 10px
}
