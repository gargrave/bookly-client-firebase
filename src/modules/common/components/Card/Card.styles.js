import styled from 'react-emotion'
import { colors, shadows } from '../../../../styles/'

export const StyledWrapper = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`
