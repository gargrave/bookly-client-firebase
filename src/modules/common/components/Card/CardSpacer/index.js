import CardSpacer from './CardSpacer'

export default CardSpacer
