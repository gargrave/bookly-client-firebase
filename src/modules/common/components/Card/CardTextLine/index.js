import CardTextLine from './CardTextLine'

export default CardTextLine
