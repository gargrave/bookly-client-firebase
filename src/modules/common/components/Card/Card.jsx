// @flow
import React from 'react'
import { any, bool, func } from 'prop-types'
import styled from 'react-emotion'

import { buildClass } from '../../../../utils/cssHelpers'

import CardHeader from './CardHeader'
import CardSpacer from './CardSpacer'
import CardTextLine from './CardTextLine'
import CardTextList from './CardTextList'

import { colors, shadows } from '../../../../styles/'

type Props = {
  children?: any,
  hoverable?: boolean,
  onClick?: Function,
}

const Styled = styled('div')`
  ${shadows.medium};

  background-color: ${colors.cardBg};
  border: solid 1px ${colors.cardBorder};
  border-radius: 5px;
  margin: 8px auto;
  max-width: 550px;
  padding: 10px 20px;
  text-align: center;

  &.hoverable:hover {
    background: ${colors.cardHoveredBg};
    cursor: pointer;
  }
`

const Card = ({ children, hoverable = false, onClick }: Props) => {
  return (
    <Styled className={buildClass({ hoverable })} onClick={onClick}>
      {children}
    </Styled>
  )
}

Card.Header = CardHeader
Card.Spacer = CardSpacer
Card.TextLine = CardTextLine
Card.TextList = CardTextList

Card.propTypes = {
  children: any,
  hoverable: bool,
  onClick: func,
}

export default Card
