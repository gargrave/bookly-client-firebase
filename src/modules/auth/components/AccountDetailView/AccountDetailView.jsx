// @flow
import React from 'react';
import { bool, func, instanceOf, oneOfType, shape, string } from 'prop-types';

import type { User } from '../../flowtypes';
import type { Profile } from '../../../profiles/flowtypes';

import styles from './AccountDetailView.css';

import {
  userBasicDetailsTextList,
  userRegDetailsTextList,
} from './helpers';

import Button from '../../../common/components/Button/Button';
import ButtonRow from '../../../common/components/ButtonRow/ButtonRow';
import Card from '../../../common/components/Card/Card';
import VerifyAccountNotice from '../VerifyAccountNotice/VerifyAccountNotice';

type Props = {
  onEditClick: Function,
  onLogoutClick: Function,
  onVerifyAccountClick: Function,
  profile: Profile,
  user: User,
  verificationEmailHasBeenSent: boolean,
};

const AccountDetailView = ({
  onEditClick,
  onLogoutClick,
  onVerifyAccountClick,
  profile,
  user,
  verificationEmailHasBeenSent,
}: Props) => {
  return (
    <div className={styles.accountDetailView}>
      <Card
        classes={['detail-card', 'account-detail-card']}
        header="My Account"
        hoverable={false}
      >

        <Card.Spacer />
        <Card.TextList textList={userBasicDetailsTextList(user, profile)} />
        <Card.Spacer />
        <Card.TextList textList={userRegDetailsTextList(user)} />
        <Card.Spacer />

        <ButtonRow>
          <Button
            onClick={onLogoutClick}
            position="left"
            text="Logout"
            type="info"
          />
          <Button
            onClick={onEditClick}
            position="right"
            text="Edit"
            type="secondary"
          />
        </ButtonRow>
      </Card>
      {!user.emailVerified &&
        <VerifyAccountNotice
          emailHasBeenSent={verificationEmailHasBeenSent}
          onSendLinkClick={onVerifyAccountClick}
        />
      }
    </div>
  );
};

AccountDetailView.propTypes = {
  onEditClick: func.isRequired,
  onLogoutClick: func.isRequired,
  onVerifyAccountClick: func.isRequired,
  user: shape({
    email: string,
    emailVerified: bool,
    lastLogin: oneOfType([
      instanceOf(Date),
      string,
    ]),
    registered: oneOfType([
      instanceOf(Date),
      string,
    ]),
  }).isRequired,
  verificationEmailHasBeenSent: bool.isRequired,
};

export default AccountDetailView;
