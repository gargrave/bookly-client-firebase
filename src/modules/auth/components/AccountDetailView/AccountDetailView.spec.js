import { profileMocks, userMocks } from '../../../../globals/mocks/';
import { ComponentBuilder } from '../../../../globals/utils/testHelpers';

import Card from '../../../common/components/Card/Card';

import AccountDetailView from './AccountDetailView';

const defaultProps = {
  onEditClick: jest.fn(),
  onLogoutClick: jest.fn(),
  onVerifyAccountClick: jest.fn(),
  profile: { ...profileMocks[0] },
  user: { ...userMocks[0] },
  verificationEmailHasBeenSent: false,
};

const builder = new ComponentBuilder(
  AccountDetailView,
  defaultProps,
);

describe('AccountDetailView', () => {
  let component;

  test('matches the snapshot', () => {
    component = builder.shallowGetComponent();
    expect(component).toMatchSnapshot();
  });

  test('renders correctly', () => {
    component = builder.shallowGetComponent();
    expect(component.find('.bookly-account-detail-view')).toHaveLength(1);
    expect(component.find(Card)).toHaveLength(1);
  });
});
