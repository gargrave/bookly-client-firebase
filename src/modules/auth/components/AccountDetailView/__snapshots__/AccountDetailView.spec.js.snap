// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailView matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccountDetailView
    onEditClick={[Function]}
    onLogoutClick={[Function]}
    onVerifyAccountClick={[Function]}
    profile={
      Object {
        "created": 2017-07-01T01:16:43.881Z,
        "firstName": "Larry",
        "lastName": "Finklestein",
        "owner": "oiw223h3qndlksadffl3ksdf",
        "updated": 2017-07-01T01:16:43.881Z,
      }
    }
    user={
      Object {
        "displayName": null,
        "email": "mockuser1@gmail.com",
        "emailVerified": true,
        "id": "h6EfCE6MrJ5IVf2ay3QsPdX5Jd23",
        "lastLogin": 2017-07-01T01:11:03.942Z,
        "registered": 2017-06-30T19:52:54.644Z,
      }
    }
    verificationEmailHasBeenSent={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          header="My Account"
        >
          <CardSpacer />
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Name",
                  "value": "Larry Finklestein",
                },
                Object {
                  "title": "Email",
                  "value": "mockuser1@gmail.com",
                },
              ]
            }
          />
          <CardSpacer />
          <CardTextList
            textList={
              Array [
                Object {
                  "title": "Registered",
                  "value": "Jun. 30, 2017, 12:52:54",
                },
                Object {
                  "title": "Last login",
                  "value": "Jun. 30, 2017, 18:11:03",
                },
              ]
            }
          />
          <CardSpacer />
          <ButtonRow>
            <Button
              onClick={[Function]}
              position="left"
              text="Logout"
              type="info"
            />
            <Button
              onClick={[Function]}
              position="right"
              text="Edit"
              type="secondary"
            />
          </ButtonRow>
        </Card>,
        false,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardSpacer />,
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Name",
                    "value": "Larry Finklestein",
                  },
                  Object {
                    "title": "Email",
                    "value": "mockuser1@gmail.com",
                  },
                ]
              }
            />,
            <CardSpacer />,
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Registered",
                    "value": "Jun. 30, 2017, 12:52:54",
                  },
                  Object {
                    "title": "Last login",
                    "value": "Jun. 30, 2017, 18:11:03",
                  },
                ]
              }
            />,
            <CardSpacer />,
            <ButtonRow>
              <Button
                onClick={[Function]}
                position="left"
                text="Logout"
                type="info"
              />
              <Button
                onClick={[Function]}
                position="right"
                text="Edit"
                type="secondary"
              />
            </ButtonRow>,
          ],
          "header": "My Account",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "textList": Array [
                Object {
                  "title": "Name",
                  "value": "Larry Finklestein",
                },
                Object {
                  "title": "Email",
                  "value": "mockuser1@gmail.com",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "textList": Array [
                Object {
                  "title": "Registered",
                  "value": "Jun. 30, 2017, 12:52:54",
                },
                Object {
                  "title": "Last login",
                  "value": "Jun. 30, 2017, 18:11:03",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Button
                  onClick={[Function]}
                  position="left"
                  text="Logout"
                  type="info"
                />,
                <Button
                  onClick={[Function]}
                  position="right"
                  text="Edit"
                  type="secondary"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onClick": [Function],
                  "position": "left",
                  "text": "Logout",
                  "type": "info",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "onClick": [Function],
                  "position": "right",
                  "text": "Edit",
                  "type": "secondary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            header="My Account"
          >
            <CardSpacer />
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Name",
                    "value": "Larry Finklestein",
                  },
                  Object {
                    "title": "Email",
                    "value": "mockuser1@gmail.com",
                  },
                ]
              }
            />
            <CardSpacer />
            <CardTextList
              textList={
                Array [
                  Object {
                    "title": "Registered",
                    "value": "Jun. 30, 2017, 12:52:54",
                  },
                  Object {
                    "title": "Last login",
                    "value": "Jun. 30, 2017, 18:11:03",
                  },
                ]
              }
            />
            <CardSpacer />
            <ButtonRow>
              <Button
                onClick={[Function]}
                position="left"
                text="Logout"
                type="info"
              />
              <Button
                onClick={[Function]}
                position="right"
                text="Edit"
                type="secondary"
              />
            </ButtonRow>
          </Card>,
          false,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardSpacer />,
              <CardTextList
                textList={
                  Array [
                    Object {
                      "title": "Name",
                      "value": "Larry Finklestein",
                    },
                    Object {
                      "title": "Email",
                      "value": "mockuser1@gmail.com",
                    },
                  ]
                }
              />,
              <CardSpacer />,
              <CardTextList
                textList={
                  Array [
                    Object {
                      "title": "Registered",
                      "value": "Jun. 30, 2017, 12:52:54",
                    },
                    Object {
                      "title": "Last login",
                      "value": "Jun. 30, 2017, 18:11:03",
                    },
                  ]
                }
              />,
              <CardSpacer />,
              <ButtonRow>
                <Button
                  onClick={[Function]}
                  position="left"
                  text="Logout"
                  type="info"
                />
                <Button
                  onClick={[Function]}
                  position="right"
                  text="Edit"
                  type="secondary"
                />
              </ButtonRow>,
            ],
            "header": "My Account",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "textList": Array [
                  Object {
                    "title": "Name",
                    "value": "Larry Finklestein",
                  },
                  Object {
                    "title": "Email",
                    "value": "mockuser1@gmail.com",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "textList": Array [
                  Object {
                    "title": "Registered",
                    "value": "Jun. 30, 2017, 12:52:54",
                  },
                  Object {
                    "title": "Last login",
                    "value": "Jun. 30, 2017, 18:11:03",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Button
                    onClick={[Function]}
                    position="left"
                    text="Logout"
                    type="info"
                  />,
                  <Button
                    onClick={[Function]}
                    position="right"
                    text="Edit"
                    type="secondary"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "onClick": [Function],
                    "position": "left",
                    "text": "Logout",
                    "type": "info",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "onClick": [Function],
                    "position": "right",
                    "text": "Edit",
                    "type": "secondary",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
